pkg/bytecode/compiler.go

```diff
@@ -130,6 +143,9 @@ func (c *Compiler) compileBinaryExpression(expr *parser.BinaryExpression) error
 	if expr.Left.Type() == parser.STRING_TYPE && expr.Right.Type() == parser.STRING_TYPE {
 		return c.compileStringBinaryExpression(expr)
 	}
+	if expr.Left.Type().Name == parser.ARRAY && expr.Right.Type().Name == parser.ARRAY && expr.Op == parser.OP_PLUS {
```

Comment:
I am unsure if this is the best way to type check arrays, for the compiler it doesn't matter what the sub-type is. It just needs to make sure that the arguments are of type `ARRAY` to know that it should output the correct opcode (and not `OpAdd` or `OpStringConcatenate`)